### ISSUE TITLE
FakeLoadExecutor should be able to stop the current running tasks manually

### ISSUE THREAD

**Issue by Cyb0rk** [msigwart/fakeload#7]
Hi there,

FakeLoadExecutor should be able to stop the current running tasks manually. If this will be implemented, it should be backported to 0.4.0 for using with Java 8.

Best regards

**Comment by msigwart** [comment]
Hey, thanks for issue. Can you elaborate a bit on what you mean with manually? Do you mean some way to programmatically stop the currently running task?

**Comment by Cyb0rk** [comment]
Correct. When a task is started via

```
FakeLoad fakeload = FakeLoads.create()
    .lasting(500, TimeUnit.SECONDS)
    .withCpu(80);
 
FakeLoadExecutor executor = FakeLoadExecutors.newDefaultExecutor(); 
executor.execute(fakeload);
```

there should be a way to stop it. Currently you have to way until the 500 seconds are passed. It would be nice to have a method like  `executor.stopFakeload();`.

**Comment by msigwart** [comment]
Okay, that makes sense. What is your exact use case if I may ask?

**Comment by Cyb0rk** [comment]
Of course:

I use the library to fake load on test environments for several tasks (loosely speaking: number calculations). On the test systems there isn't much workload, so the tasks would be over after a few seconds. I'll use your library to let the "work" run over several minutes / hours.

But those task could be stopped manually via the UI. Currently, on test systems I have to quit the whole program to stop the "fake" thread.

**Comment by msigwart** [comment]
Right. I'll see if I can implement something next week. But I'm not sure if I'll be able to backport for Java 8 though. Switching to the latest Java version is not an option for you?

**Comment by Cyb0rk** [comment]
Thanks a lot!

**Comment by msigwart** [comment]
Took me a bit longer to get to it, but you can now cancel load executions:
```java
FakeLoad fakeload = FakeLoads.create()
    ...
 
FakeLoadExecutor executor = FakeLoadExecutors.newDefaultExecutor();

// Execute FakeLoad synchronously
executor.execute(fakeload);

// Execute FakeLoad asynchronously
Future<Void> execution = executor.executeAsync();
execution.get(); // wait for completion
execution.cancel(); // cancel execution
``` 
If this is still relevant for you, the update is available in version 0.6.0. Thanks again for the issue, will close it for now!
